PP-382: Support for Postgres-9.3.12 on Windows platform

gajendrasharmagit opened the pull request:
<!--- Please review your changes in preview mode -->
<!--- Provide a general summary of your changes in the Title above -->

#### Issue-ID
<!--- Replace XXXX below (2 places) with the actual JIRA id -->
* **[PP-382](https://pbspro.atlassian.net/browse/PP-XXXX)**

#### Problem description
* PBS needs to support current Postgres version 9.3.12.

#### Cause / Analysis
* PBS needs to support current Postgres version 9.3.12. 

#### Solution description
* printjob_svr no longer needs to undef pid_t as there is no definition conflict of pid_t, with Postgres 9.3.12.
* .gitignore is updated with a few more auto generated files. Also, last submission accidentally removed a few auto-generated files on UNIX. Re-added.

#### Checklist:
<!--- Use the preview button to see the checkboxes/links properly. -->
<!--- Go over the following points, and put an `x` (without spaces around it) in the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [x] I have added  **manual test(s) to the Jira ticket and explained why PTL is not appropriate** for this case.
- [ ] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to the Jira ticket** as evidence of testing/verification.


__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

bremanandjk commented:
Thanks for your explanation, I sign-off on these changes

subhasisb commented on .gitignore:
```diff
@@ -194,8 +194,13 @@ src/lib/Libpbs/ecl_resc_def_all.c
+src/lib/Libpython/pbs_ifl_wrap.c
 
 #Generated source files - Windows
+win_configure/include/pbs_version_win.h
```
You are adding pbs_version_win.h to .gitignore; however this file is actually still part of the repository. If this is indeed an autogenerated file, then it should be removed from the repo

gajendrasharmagit commented:
Subhasis and Bream,
I have removed the legacy pbs_version_win.h. I have updated the template pbs_version.h.in and pre-build event script gen_pbs_version.bat. Please have a look.

subhasisb commented:
Code looks good. I sign off.

Can you kindly attached a fresh windows buildlog which shows that it actually builds on windows from scratch (ie, without a backup/vestigial pbs_version.h lying around)?

gajendrasharmagit commented:
Thanks. Updated the ticket with fresh build logs on Windows without vestigial pbs_version.h.

subhasisb commented:
All in place. Merging code.

gajendrasharmagit commented:
Thanks.
